### PR TITLE
feat(controls): Add react controls for mp4 viewer

### PR DESCRIPTION
**Pull request by jstoffan**
Todo
- [x] Cross-browser testing

### CODE REVIEW

File: src/lib/viewers/media/__tests__/MP3Viewer-test.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /* eslint-disable no-unused-expressions */
 import BaseViewer from '../../BaseViewer';
 import MP3Viewer from '../MP3Viewer';
+import MP3ControlsRoot from '../MP3ControlsRoot';
```
[the remark this answers is not in the document]
Comment (jstoffan) [reply src/lib/viewers/media/__tests__/MP3Viewer-test.js:4]: Hm, the sorter plugin I use puts capital letters above lowercase, as does JS. Going to leave a couple of these as-is.

File: src/lib/viewers/media/MP4.scss [hunk 2]
```diff
@@ -34,4 +34,12 @@
             opacity: 1;
         }
     }
+
+    .bp-VideoControlsRoot {
+        position: absolute;
+        right: 0;
+        bottom: -1px;
```
Comment (ConradJChan) [inline src/lib/viewers/media/MP4.scss:41]: What is this `-1px` positioning for?

File: src/lib/viewers/media/VideoBaseViewer.js [hunk 3]
```diff
@@ -88,6 +97,22 @@ class VideoBaseViewer extends MediaBaseViewer {
         super.loadUI();
     }
 
+    /**
+     * @inheritdoc
+     */
+    loadUIReact() {
+        super.loadUIReact();
+
+        this.controls = new ControlsRoot({
```
Comment (ConradJChan) [inline src/lib/viewers/media/VideoBaseViewer.js:106]: Do we call `this.controls.destroy()` somewhere?

Comment (jstoffan) [reply src/lib/viewers/media/VideoBaseViewer.js:106]: We do for MP3, but not for MP4/video. Good catch.

File: src/lib/viewers/media/MP4.scss (hunk 2 above)
Comment (jstoffan) [reply src/lib/viewers/media/MP4.scss:41]: I've noticed there can be a 1px of video showing through below the controls, so this allows it to be covered up by the semi-transparent controls background.

File: src/lib/viewers/controls/controls-root/ControlsRoot.tsx [hunk 4]
```diff
@@ -21,9 +24,13 @@ export default class ControlsRoot {
         show: noop,
     };
 
-    constructor({ containerEl, fileId }: Options) {
+    handleHide: () => void;
+
+    handleShow: () => void;
+
+    constructor({ className = 'bp-ControlsRoot', containerEl, fileId, onHide = noop, onShow = noop }: Options) {
```
[the remark this answers is not in the document]
Comment (jstoffan) [reply src/lib/viewers/controls/controls-root/ControlsRoot.tsx:31]: It needs to be replaced for now. We can re-evaluate this later if it becomes an issue.

**Review by ConradJChan — APPROVED** [review]

**Pull request merged**